pkgdb/include/flox/pkgdb/write.hh

```diff
@@ -124,6 +124,19 @@ public:
     : PkgDb( flake, genPkgDbName( flake.getFingerprint() ).string() )
   {}
 
+  /* Connecting and locking */
+
+  /**
+   * @brief Tries to connect to the database, acquiring an exclusive lock on it.
```

Comment:
That's not exactly how it works, I'll add a better comment in the implementation. With `locking_mode = exclusive` the first time you write to the database an exclusive lock is obtained and held until the connection is closed ([source](https://www.sqlite.org/pragma.html#pragma_locking_mode)). So what's happening here is that we try to acquire the lock, blocking until we're able to, and once that retry loop exits we should be the only process able to write to the database. In short, after this retry loop exits we have the lock until we close the database connection by explicitly calling `closeDbReadWrite`.

The `BEGIN EXCLUSIVE TRANSACTION` statements that you see elsewhere make it so that no one else can _read_ from the database during the duration of the transaction. So there's a brief period between us acquiring the write lock and starting the transaction inside `scrapePrefix` that someone could try to read, but you only initialize a read-only database in `PkgDbInput::init` after you initialize the read-write database, which will block while someone else has the write lock. Practically speaking, very unlikely to happen.

Reply:
Thanks that was helpful, I definitely didn't get it before. Might be nice to add a comment with
```
So there's a brief period between us acquiring the write lock and starting the transaction inside scrapePrefix that someone could try to read, but you only initialize a read-only database in PkgDbInput::init after you initialize the read-write database, which will block while someone else has the write lock. Practically speaking, very unlikely to happen.
```

One other thing I still don't quite get - when you're acquiring the write lock in `connect`, why do you create an exclusive transaction?